src/main/handlers/transcriptionHandler.ts

```diff
@@ -13,8 +14,21 @@ const sleep: (n: number) => Promise<void> = (n) =>
   new Promise((resolve) => setTimeout(resolve, n * 1000));
 
 const handleTranscription: (
-  fileName: string
-) => Promise<Transcription> = async () => {
+  filePath: string
+) => Promise<Transcription> = async (filePath: string) => {
+  // TODO: replace hard coded media path with parameter passed in
+  const pathToSaveMedia = path.join(
+    process.cwd(),
+    'assets',
+    'audio',
+    'audio.wav'
+  );
+  const audioDurationPromise = getAudioDurationInSeconds(pathToSaveMedia)
```

Comment:
Rather than storing as a promise that is immediately awaited, you can just await here:

```
const audioDuration = (await getAudioDurationInSeconds(pathToSaveMedia)) || 0;
```

Reply:
Was thinking the promise can async complete itself while the transcription transcribes, but it's probs not a huge performance difference regardless

Reply:
Since there is no large hit on performance i think awaiting the promise when the function is called is the way to go.

One less thing to keep track of, and one less variable.

Reply:
I see what you're saying - if you want to go this route you could use Promise.all, it's a bit cleaner. But I agree with Mutaz, the "get duration" operation should be very fast so I would go with the simpler approach

Reply:
Ok, I've gone with the immediate await in the latest commit